README.md

```diff
@@ -20,62 +91,429 @@ Add the following to your `pubspec.yaml` file:
-        // open url in a webview
-      },
-      style: {
-        "div": Style(
-          block: Block(
-            margin: EdgeInsets.all(16),
-            border: Border.all(width: 6),
-            backgroundColor: Colors.grey,
-          ),
-          textStyle: TextStyle(
-            color: Colors.red,
-          ),
-        ),
+Widget html = Html(
+  data: """<div>
+        <h1>Demo Page</h1>
+        <p>This is a fantastic product that you should buy!</p>
+        <h3>Features</h3>
+        <ul>
+          <li>It actually works</li>
+          <li>It exists</li>
+          <li>It doesn't cost much!</li>
+        </ul>
+        <!--You can pretty much put any html in here!-->
+      </div>""",
+);
+```
+
+### onLinkTap:
+
+A function that defines what the widget should do when a link is tapped.
+
+#### Example Usage - onLinkTap:
+
+```dart
+Widget html = Html(
+  data: """<p>
+   Linking to <a href='https://github.com'>websites</a> has never been easier.
+  </p>""",
+  onLinkTap: (String url) {
+    //open URL in webview, or launch URL in browser, or any other logic here
+  }
+);
+```
+
+### customRender:
+
+A powerful API that allows you to customize everything when rendering a specific HTML tag. This means you can add support for HTML elements that aren't supported natively. You can also make up your own custom tags in your HTML!
+
+`customRender` accepts a `Map<String, CustomRender>`. The `CustomRender` type is a function that requires a `Widget` to be returned. It exposes `RenderContext`, the `Widget` that would have been rendered by `Html` without a `customRender` defined, the `attributes` of the HTML element as a `Map<String, String>`, and the HTML element itself as `Element`.
+
+To use this API, set the key as the tag of the HTML element you wish to provide a custom implementation for, and create a function with the above parameters that returns a `Widget`.
+
+#### Example Usages - customRender:
+1. Simple example - rendering custom HTML tags
+<details><summary>View code</summary>
+
+```dart
+Widget html = Html(
+  data: """
+  <h3>Display bird element and flutter element <bird></bird></h3>
+  <flutter></flutter>
+  <flutter horizontal></flutter>
+  """,
+  customRender: {
+      "bird": (RenderContext context, Widget child, Map<String, String> attributes, _) {
+        return TextSpan(text: "🐦");
       },
-      onImageTap: (src) {
-        // Display the image in large form.
+      "flutter": (RenderContext context, Widget child, Map<String, String> attributes, _) {
+        return FlutterLogo(
+          style: (attributes['horizontal'] != null)
+              ? FlutterLogoStyle.horizontal
+              : FlutterLogoStyle.markOnly,
+          textColor: context.style.color,
+          size: context.style.fontSize.size * 5,
+        );
       },
+    },
+);
+```
+</details>
+
+2. Complex example - rendering an `iframe` differently based on whether it is an embedded youtube video or some other embedded content
```

Comment:
It's indeed great to have a more extensive example. And the case here (render an iframe or embed YouTube video) is a good example. Only the connection checking I'm not sure of. That doesn't really explain how to use flutter_html in my opinion.

Reply:
Came up with a (hopefully) better example using iframe and video, let me know your thoughts.